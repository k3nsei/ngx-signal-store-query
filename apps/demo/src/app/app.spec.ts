import { provideHttpClient } from '@angular/common/http';
import { provideHttpClientTesting } from '@angular/common/http/testing';
import { type ComponentFixture, TestBed } from '@angular/core/testing';

import { provideTanStackQuery, QueryClient } from '@tanstack/angular-query-experimental';

import { App } from './app';

describe('GIVEN AppComponent', () => {
  let fixture: ComponentFixture<App>;
  let component: App;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [App],
      providers: [
        provideHttpClient(),
        provideHttpClientTesting(),
        provideTanStackQuery(
          new QueryClient({
            defaultOptions: {
              queries: {
                refetchOnWindowFocus: false,
                retry: false,
                staleTime: 0,
              },
            },
          }),
        ),
      ],
    });

    fixture = TestBed.createComponent(App);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  describe('WHEN component was initialized', () => {
    it('THEN component instance should be truthy', () => {
      expect(component).toBeTruthy();
    });

    it('THEN component should be rendered', () => {
      expect(fixture.nativeElement).toMatchSnapshot();
    });
  });
});
