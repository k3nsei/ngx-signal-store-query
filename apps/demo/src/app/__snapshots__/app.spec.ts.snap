// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GIVEN AppComponent > WHEN component was initialized > THEN component should be rendered 1`] = `
<div
  id="root1"
  ng-version="20.3.7"
>
  <ssq-counter>
    <output
      aria-label="Current count"
    >
      0
    </output>
    <button
      aria-label="Increase"
      class="mdc-fab mat-mdc-fab-base mat-mdc-fab mat-mdc-button-base mat-accent"
      mat-fab=""
      mat-ripple-loader-class-name="mat-mdc-button-ripple"
      mat-ripple-loader-uninitialized=""
      title="Increase"
      type="button"
    >
      <span
        class="mat-mdc-button-persistent-ripple mdc-fab__ripple"
      />
      <mat-icon
        aria-hidden="true"
        class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
        data-mat-icon-name="exposure_plus_1"
        data-mat-icon-type="font"
        fonticon="exposure_plus_1"
        role="img"
      />
      <span
        class="mdc-button__label"
      />
      <span
        class="mat-focus-indicator"
      />
      <span
        class="mat-mdc-button-touch-target"
      />
    </button>
    <button
      aria-label="Decrease"
      class="mdc-fab mat-mdc-fab-base mat-mdc-fab mat-mdc-button-base mat-accent"
      mat-fab=""
      mat-ripple-loader-class-name="mat-mdc-button-ripple"
      mat-ripple-loader-uninitialized=""
      title="Decrease"
      type="button"
    >
      <span
        class="mat-mdc-button-persistent-ripple mdc-fab__ripple"
      />
      <mat-icon
        aria-hidden="true"
        class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
        data-mat-icon-name="exposure_neg_1"
        data-mat-icon-type="font"
        fonticon="exposure_neg_1"
        role="img"
      />
      <span
        class="mdc-button__label"
      />
      <span
        class="mat-focus-indicator"
      />
      <span
        class="mat-mdc-button-touch-target"
      />
    </button>
  </ssq-counter>
  <ssq-github-repos>
    <mat-progress-bar
      aria-valuemax="100"
      aria-valuemin="0"
      class="mat-mdc-progress-bar mdc-linear-progress busy-indicator mat-primary mdc-linear-progress--animation-ready mdc-linear-progress--indeterminate"
      mode="indeterminate"
      role="progressbar"
      tabindex="-1"
    >
      <div
        aria-hidden="true"
        class="mdc-linear-progress__buffer"
      >
        <div
          class="mdc-linear-progress__buffer-bar"
          style="flex-basis: 100%;"
        />
        <!--container-->
      </div>
      <div
        aria-hidden="true"
        class="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
        style="transform: scaleX(1);"
      >
        <span
          class="mdc-linear-progress__bar-inner"
        />
      </div>
      <div
        aria-hidden="true"
        class="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
      >
        <span
          class="mdc-linear-progress__bar-inner"
        />
      </div>
    </mat-progress-bar>
    <!--container-->
    <mat-list
      aria-disabled="false"
      class="mat-mdc-list mat-mdc-list-base mdc-list list"
    >
      <div
        class="mat-mdc-subheader mdc-list-group__subheader"
        mat-subheader=""
      >
        Google Repositories
      </div>
      <mat-list-item
        aria-disabled="false"
        class="mat-mdc-list-item mdc-list-item list-item skeleton mdc-list-item--with-leading-icon mdc-list-item--with-two-lines"
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon notranslate mat-mdc-list-item-icon list-item__icon material-icons mat-ligature-font mat-icon-no-color mdc-list-item__start"
          data-mat-icon-type="font"
          matlistitemicon=""
          role="img"
        >
          folder
        </mat-icon>
        <span
          class="mdc-list-item__content"
        >
          <div
            class="mat-mdc-list-item-title mdc-list-item__primary-text list-item__title"
            matlistitemtitle=""
          >
            <ssq-skeleton-text
              class="skeleton-text-animated"
            >
               
            </ssq-skeleton-text>
          </div>
          <div
            class="mat-mdc-list-item-line mdc-list-item__secondary-text list-item__description"
            matlistitemline=""
          >
            <ssq-skeleton-text
              class="skeleton-text-animated"
              inlinesize="12rem"
              style="--ssq-skeleton-text-size-x: 12rem;"
            >
               
            </ssq-skeleton-text>
          </div>
          <span
            class="mat-mdc-list-item-unscoped-content"
          />
        </span>
        <div
          class="mat-focus-indicator"
        />
      </mat-list-item>
      <!--container-->
      <mat-list-item
        aria-disabled="false"
        class="mat-mdc-list-item mdc-list-item list-item skeleton mdc-list-item--with-leading-icon mdc-list-item--with-two-lines"
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon notranslate mat-mdc-list-item-icon list-item__icon material-icons mat-ligature-font mat-icon-no-color mdc-list-item__start"
          data-mat-icon-type="font"
          matlistitemicon=""
          role="img"
        >
          folder
        </mat-icon>
        <span
          class="mdc-list-item__content"
        >
          <div
            class="mat-mdc-list-item-title mdc-list-item__primary-text list-item__title"
            matlistitemtitle=""
          >
            <ssq-skeleton-text
              class="skeleton-text-animated"
            >
               
            </ssq-skeleton-text>
          </div>
          <div
            class="mat-mdc-list-item-line mdc-list-item__secondary-text list-item__description"
            matlistitemline=""
          >
            <ssq-skeleton-text
              class="skeleton-text-animated"
              inlinesize="12rem"
              style="--ssq-skeleton-text-size-x: 12rem;"
            >
               
            </ssq-skeleton-text>
          </div>
          <span
            class="mat-mdc-list-item-unscoped-content"
          />
        </span>
        <div
          class="mat-focus-indicator"
        />
      </mat-list-item>
      <!--container-->
      <mat-list-item
        aria-disabled="false"
        class="mat-mdc-list-item mdc-list-item list-item skeleton mdc-list-item--with-leading-icon mdc-list-item--with-two-lines"
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon notranslate mat-mdc-list-item-icon list-item__icon material-icons mat-ligature-font mat-icon-no-color mdc-list-item__start"
          data-mat-icon-type="font"
          matlistitemicon=""
          role="img"
        >
          folder
        </mat-icon>
        <span
          class="mdc-list-item__content"
        >
          <div
            class="mat-mdc-list-item-title mdc-list-item__primary-text list-item__title"
            matlistitemtitle=""
          >
            <ssq-skeleton-text
              class="skeleton-text-animated"
            >
               
            </ssq-skeleton-text>
          </div>
          <div
            class="mat-mdc-list-item-line mdc-list-item__secondary-text list-item__description"
            matlistitemline=""
          >
            <ssq-skeleton-text
              class="skeleton-text-animated"
              inlinesize="12rem"
              style="--ssq-skeleton-text-size-x: 12rem;"
            >
               
            </ssq-skeleton-text>
          </div>
          <span
            class="mat-mdc-list-item-unscoped-content"
          />
        </span>
        <div
          class="mat-focus-indicator"
        />
      </mat-list-item>
      <!--container-->
      <!--container-->
      <!--container-->
    </mat-list>
    <!--container-->
  </ssq-github-repos>
</div>
`;
